sno-setup.sh

```diff
@@ -39,15 +39,15 @@ else
   sshKeyFile=~/.ssh/id_rsa.pub
 fi
 
-if $(sudo virsh list --all | grep "\s${hostname}\s" &> /dev/null); then
-  echo "Found an already existing mini-agent instance, please run sno-cleanup.sh to remove it"
+if [ -d "${assets_dir}" ] || sudo virsh list --all --name | grep -q "${hostname}"; then
+  echo "Found existing miniagent state, please run sno-cleanup.sh first"
   exit 1
 fi
 
 start=$(date +%s)
 
 ### 2. Create a temporary working folder and make it accessible to the qemu and current user
-mkdir -p ${assets_dir}
```

Comment:
Not sure about the reason to drop `-p`?

Reply:
Because if we now always expect a clean state (as checked by the if-condition higher up), we expect that directory to not exist. If it does, that means there's a bug somewhere.